### PR TITLE
Add weather effect snow.

### PR DESCRIPTION
**Pull request by brettpenzer123**
This should be my last pull request for this feature. Now you can shout again.

### CODE REVIEW

**Comment by pizza2004** [comment]
This has been an interesting saga to watch.  Hopefully this one just works. :P

**Comment by brettpenzer123** [comment]
I know. I'm just getting used to Git. Thank you for those that are willing to explain (no one)

**Comment by brettpenzer123** [comment]
Wow and it fails again.

**Comment by pizza2004** [comment]
It only failed because the Xcode project wasn't updated.  Hold on let me update that for you.

File: src/drawing/drawing.c [hunk 1]
```diff
@@ -476,6 +479,59 @@ void gfx_draw_rain(int left, int top, int width, int height, sint32 x_start, sin
 	}
 }
 
+// Draw_Snow
```
Comment (IntelOrca) [inline src/drawing/drawing.c]: This comment is unnecessary because the function name already says "draw_snow"

File: src/drawing/drawing.c [hunk 2]
```diff
@@ -503,6 +559,30 @@ void redraw_rain()
 	}
 }
 
+// Redraw_Snow
```
Comment (IntelOrca) [inline src/drawing/drawing.c]: Again, unnecessary comment

File: src/world/climate.c [hunk 3]
```diff
@@ -26,6 +26,9 @@
 #include "../util/util.h"
 #include "climate.h"
 
+int SnowLevel;
+int NextSnowLevel;
```
Comment (IntelOrca) [inline src/world/climate.c]: We always prefix global variables with a small g, usually then followed by the related module. So in this case they should be called:
```
gClimateSnowLevel
gClimateNextSnowLevel
```

**Comment by pizza2004** [comment]
It appears there's still a build issue on OS X due to you having the snow level variable defined twice.

**Comment by brettpenzer123** [comment]
I'll change the variable in snow.c to see if that works.

**Comment by brettpenzer123** [comment]
If its a global variable why can't `snow.c` get access to it from `climate.c`

**Comment by pizza2004** [comment]
You need to define it in climate.h probably.

**Comment by brettpenzer123** [comment]
And have snow.c and climate.c access it from there.

**Comment by pizza2004** [comment]
extern uint16 gClimateLightningFlash; Under this add extern uint16 gClimateSnowLevel;

**Comment by brettpenzer123** [comment]
I'll reopen this in a second just making sure its all ready

File: src/world/climate.c [hunk 4]
```diff
@@ -24,7 +24,7 @@
 #include "../localisation/date.h"
 #include "../scenario.h"
 #include "../util/util.h"
-#include "climate.h"
+#include "../world/climate.h"
```
Comment (IntelOrca) [inline src/world/climate.c]: Why have you changed this? All are already in the world directory.

Comment (brettpenzer123) [inline src/world/climate.c]: That was stopping me from calling the g variable within that file.

**Comment by brettpenzer123** [comment]
Better?

**Comment by duncanspumpkin** [comment]
Stop closing and opening fresh PRs. There is no reason at all. It makes it very hard to follow the discussion and spams my email.

**Comment by brettpenzer123** [comment]
I closed this and reopened it you.should be thankful. Why do you all moan?

**Comment by mrtnptrs** [comment]
OK, firstly you work against the System (Said you made something without Sharing the code, but no one knows exactly why.) And now you finally Made the pull request, you Close and open it, which makes no sense as we have tags Here to Mark WIP pull request. And when we say this to you, you say we moan?! Sorry, it's very good that you want to Help is, but forgive me when I think your behavior is a bit odd...

**Comment by Gymnasiast** [comment]
Please, don't let this escalate.

Anyway, we use the Github workflow, and for a good reason: it allows us to keep track of reported issues, chances to the code, discussion, and pull requests. If people divert from that it gets very hard to keep track of everything, especially when it's more than one. That why we ask you to stick to that and not doing it your own way - though it has to be said that the Github workflow isn't hard to accommodate into your own way of working, as it isn't very intrusive. If you have anyway questions about the workflow we are happy to answer them on Gitter.

**Comment by brettpenzer123** [comment]
@Gymnasiast Thanks.

**Comment by brettpenzer123** [comment]
Decided that it was causing to many issues. So I'm going to just update the `rain.c` with snow.

File: src/rct2/S6Importer.cpp [hunk 5]
```diff
@@ -425,6 +425,7 @@ extern "C"
         // #2407: Resetting screen time to not open a save prompt shortly after loading a park.
         gScreenAge = 0;
         gLastAutoSaveTick = SDL_GetTicks();
+		if (gClimateCurrentWeather == WEATHER_SNOW) { climate_force_weather(WEATHER_SNOW); }
```
Comment (brettpenzer123) [inline src/rct2/S6Importer.cpp]: Don't know if this could be loaded somewhere else other then the importer.

**Comment by brettpenzer123** [comment]
This now fails due to OPENRCT2/develop being forward then my develop branch. I can't merge the two because of changes made by #3658

**Comment by janisozaur** [comment]
You shouldn't merge anyway. Rebase & fix conflicts.

**Comment by brettpenzer123** [comment]
Yeah. I should have branched it properly. Obviously didn't. I've tried rebasing it but the conflict still happens.


Get Outlook for Android






On Thu, May 19, 2016 at 2:17 PM -0700, "Michał Janiszewski" <notifications@github.com> wrote:





You shouldn't merge anyway. Rebase & fix conflicts.
Can you also squash your commits so you don't add, then revert, then revert, then add again?

---

**Comment by brettpenzer123** [comment]
and won't let me commit after the conflict has been fixed.


Get Outlook for Android






On Thu, May 19, 2016 at 2:17 PM -0700, "Michał Janiszewski" <notifications@github.com> wrote:





You shouldn't merge anyway. Rebase & fix conflicts.
Can you also squash your commits so you don't add, then revert, then revert, then add again?

---

**Comment by mrtnptrs** [comment]
Not again..... why not asking for Help instead of closing it again? 😅

**Comment by brettpenzer123** [comment]
I decided to just remove it... It had way to many problems.

**Comment by brettpenzer123** [comment]
Good Luck on the project.

**Comment by mrtnptrs** [comment]
And you even deleted everything So we are unable to fix the problems ourselves and eventually merge it?!

**Comment by brettpenzer123** [comment]
I will post the code when I get round to it.

I've been contacted about (ID@XBOX) which is absolutely great news my end. :dizzy_face:
